### DISCUSSION TITLE
reTerminal DM Getting Started | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#182] [Q&A] [answered]
# reTerminal DM Getting Started | Seeed Studio Wiki

reTerminal DM Getting Started

https://wiki.seeedstudio.com/reterminal-dm/

**Comment by Owizmi** [comment]
How I get more detailed datasheets for IO connector.

**Reply by themzlab** [reply]
I have the same question.  I looked in the product datasheet and the Wiki and have not found a block diagram nor pinout for the product.

**Reply by Ainsfei** [reply]
The user manual will be released this week, including block diagram and IO connector.

**Reply by Ainsfei** [reply, marked as answer]
Hi, please check the user manul https://files.seeedstudio.com/wiki/reTerminalDM/reTerminal-DM-User-Manual.pdf

**Comment by MartinRaSt** [comment]
Hi, 
How can I add SSD to this device. Datasheet remarks it is optional 2280B but how can I attach it?
Thank You

**Reply by Ainsfei** [reply]
The user manual will be released this week. SSD is optional, to use it need to istall an PCIe extension card.

**Reply by Ainsfei** [reply]
Hi, please check the user manual https://files.seeedstudio.com/wiki/reTerminalDM/reTerminal-DM-User-Manual.pdf

**Comment by tylerseitz** [comment]
Is there CAD (.stp) files available for the Reterminal DM?

**Reply by LJ-Hao** [reply]
Here is the reTerminal DM CAD files [link](https://files.seeedstudio.com/wiki/reTerminalDM/reTerminal-DM-Structural-Design-File.stp)

**Comment by themzlab** [comment]
Can the digital outputs be set to open-collector?  if possible, would configure it to either sink current or be an open circuit.  Otherwise, are there additional details available regarding the electrical specification of the digital outputs?

**Reply by bclswl0827** [reply]
Unfortunately, it cannot set to open-collector, since the output of `EXT_OUT*` is driven by an NPN transistor, while the collector is connected to `DO_VDD` via a fuse and phototransistor. The `EXT_OUT*` itself is an output with a voltage source, so it cannot be an open-collector configuration.

![image](https://github.com/user-attachments/assets/dd5991e4-1c84-4343-85c7-cf0b1100fa59)

P.S. you can get the schematic PDF [here](https://files.seeedstudio.com/wiki/reTerminalDM/reTerminal_DM_schematic_design_files.zip).

**Comment by j-broome** [comment]
Need assistance with the LCD. When I do a reboot, the screen stays blank sometimes. Any help is much appreciated.

**Reply by bclswl0827** [reply]
Hi, as of https://github.com/Seeed-Studio/pi-gen-expand/commit/8affb3b, the fix for this issue has been integrated into the system service. You may consider downloading the image from https://github.com/Seeed-Studio/pi-gen-expand/releases/latest and check if the problem still exists, or simply add https://github.com/Seeed-Studio/pi-gen-expand/blob/main/dev_config/reTerminal-DM/files/dsi_fix.sh to system service manually.

**Reply by j-broome** [reply]
Is there a typo in the dsi_fix.sh file? 
It shows the same line twice. I have the service running but still see issue.

**Reply by j-broome** [reply]
Okay so I flashed with the [Raspbian-reTerminal-DM-arm64.zip](https://github.com/Seeed-Studio/pi-gen-expand/releases/download/v1.1.0/Raspbian-reTerminal-DM-arm64.zip) to the unit. Rebooted 10 times and never saw any issues. Ran sudo apt update and sudo apt upgrade -y, then rebooted and screen is blank.

**Reply by bclswl0827** [reply]
Reproduced.

After restarting, sometimes the screen goes blank, however the /sys/devices/platform/soc/fe700000.dsi/fe700000.dsi.0/dsi_state still returns ok.

When the screen goes blank, we can try to enable the screen by reloading the driver with the following commands

```
$ sudo rmmod ili9881d
$ sudo modprobe ili9881d
```

And now we've been able to control the screen, e.g.

```
$ sudo su
$ cat /dev/urandom > /dev/fb0
```

And according to the kernel message, now we can conclude that the problem lies in the Linux DRM framework.

It may be that after the DRM-related package(s) was updated, the bug of the function `ili9881d_get_modes` in the screen driver was triggered.

I have transferred this to the R&D staff - the next work involves modification on ili9881d driver.

**Comment by paulflake** [comment]
Hello,
I tried to connect the reTerminal DM to a barcode scanner which emulates a RS232 interface via USB Type-A. Unfortunately, the scanner is not recognized as a /dev/tty device (/dev/ttyACM0) on the reTerminal DM. If I use a normal Raspberry Pi board, the device is able to find the scanner as /dev/ttyACM0. I also tried to activate the serial interface within the raspi-config without success. 

Is it possible to solve the problem without using the IO-pins?
Any help would be appreciated.

**Reply by themzlab** [reply]
use the command dmesg to see what has happened when you plug in and then unplug the scanner.  You can also use sudo dmesg --clear before you do this and it will be easier to see the response when plugging in.

**Comment by XD-QIN** [comment]
After updating to bookworm and installing the driver, the display won't go to sleep like bullseye. Anyway to resolve this issue?

**Reply by nfs0619** [reply]
Hello, thank you for your feedback. I recommend reaching out to our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)** and sharing relevant information such as operation logs. They will be able to provide you with more detailed and timely guidance.

**Comment by torchia-tech** [comment]
Does this have a panel layout document for cutout of electrical panel?

**Comment by j-broome** [comment]
I made one from the provided CAD file I can email to you.

***@***.***

--
Jordan Broome
Electrical Engineer

Cell:575-760-0497 ­Email: ***@***.*** ***@***.***)

Sent from [Proton Mail](https://proton.me/mail/home) for iOS.

-------- Original Message --------

**Reply by torchia-tech** [reply]
Please do. Thank you

**Comment by j-broome** [comment]
I don't see your email on this chain so if you email me I can reply with the file.

--
Jordan Broome
Electrical Engineer

Cell:575-760-0497 ­Email: ***@***.*** ***@***.***)

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

**Reply by torchia-tech** [reply]
Please try now should be available in my profile - Github keeps adding asterisks to it.

**Comment by torchia-tech** [comment]
Please send to this email:

***@***.***

**Comment by j-broome** [comment]
It's hiding the email addresses in the chain. I tried to attach the files that hopefully help you.

Respectfully,

--
Jordan Broome
Electrical Engineer

Cell:575-760-0497 ­Email: ***@***.*** ***@***.***)

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

**Reply by torchia-tech** [reply]
Please try now should be available in my profile - Github keeps adding asterisks to it.
